flux/src/flagger/canaries.tsx

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import { Box } from '@mui/material';
+import { K8s } from '@kinvolk/headlamp-plugin/lib';
+import { Table, SectionBox, Loader, Link } from '@kinvolk/headlamp-plugin/lib/components/common';
+import { Icon } from '@iconify/react';
+import CanaryStatus from './canarystatus';
+import { DeploymentProgress } from './deploymentprogress';
+import FlaggerAvailabilityCheck from './availabilitycheck';
+
+export default function Canaries() {
+    const [canary, error] = K8s.ResourceClasses.CustomResourceDefinition.useGet('canaries.flagger.app');
```

Comment:
Maybe this could be abstracted into a useCanary helper, since you have it in 2 places.

Reply:
updated